sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/Framework/AuthenticationEventRequest.cs

```diff
@@ -36,36 +36,64 @@ public abstract class AuthenticationEventRequest<TResponse, TData> : Authenticat
 
         /// <summary>Validates the response and creates the IActionResult with the json payload based on the status of the request.</summary>
         /// <returns>IActionResult based on the EventStatus (UnauthorizedResult, BadRequestObjectResult or JsonResult).</returns>
-        public async override Task<AuthenticationEventResponse> Completed()
+        public override AuthenticationEventResponse Completed()
         {
             try
             {
                 if (RequestStatus == RequestStatusType.Failed || RequestStatus == RequestStatusType.ValidationError)
                 {
-                    Response.MarkAsFailed(new Exception(string.IsNullOrEmpty(StatusMessage) ? AuthenticationEventResource.Ex_Gen_Failure : StatusMessage), true);
+                    return Failed(new AuthenticationEventTriggerRequestValidationException(string.IsNullOrEmpty(StatusMessage) ? AuthenticationEventResource.Ex_Gen_Failure : StatusMessage));
                 }
                 else if (RequestStatus == RequestStatusType.TokenInvalid)
                 {
-                    Response.MarkAsUnauthorized();
+                    Unauthorized();
                 }
             }
             catch (Exception ex)
             {
-                return await Failed(ex, true).ConfigureAwait(false);
+                return Failed(ex);
             }
 
             return Response;
         }
 
-        internal override Task<AuthenticationEventResponse> Failed(Exception exception, bool internalError)
+        /// <summary>
+        /// Sets the response statuscode, reason phrase and body when we want to return failed.
+        /// </summary>
+        /// <param name="exception">The exception that is thrown</param>
+        /// <returns>An authenticationeventresponse task</returns>
+        public override AuthenticationEventResponse Failed(Exception exception)
         {
             if (Response == null)
             {
                 Response = new TResponse();
             }
 
-            Response.MarkAsFailed(exception, internalError);
-            return Task.FromResult<AuthenticationEventResponse>((TResponse)Response);
+            if (exception is AuthenticationEventTriggerValidationException ex)
+            {
+                Response.StatusCode = ex.ExceptionStatusCode;
+                Response.ReasonPhrase = ex.ReasonPhrase;
+            }
+            else
+            {
+                Response.StatusCode = System.Net.HttpStatusCode.InternalServerError;
+                Response.ReasonPhrase = "Internal Server Error";
+            }
+
+            Response.Body = Helpers.GetFailedResponsePayload(exception);
+
+            return Response;
+        }
+
+        /// <summary>
+        /// Sets the response statuscode to unauthorized and the reasonphrase and body to empty.
+        /// </summary>
+        private void Unauthorized()
```

Comment:
Let's call it `MakeResponseUnauthorized`, so it is clear from the name what method is doing

Reply:
Updated